### ISSUE TITLE
tf.train.shuffle_batch does not preserve order

### ISSUE THREAD

**Issue by conan7882** [conan7882/image-classification-tensorflow#1]
The output order is 4, 5, 6,..., n, 0, 1, 2,3
